### PR TITLE
System representation common component and url update : Fix #307

### LINKED ISSUE #307: [SystemRepresentation] Make use of common components and URL Update
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

- [x] The SystemRepresentation should use common components created for filtering. Selection inside those component should result to subscription to use reactive pattern
- [x] When selecting any filter, this selection should be reflected inside the URL

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #307 
- The SystemRepresentation uses common components created for filtering. 
- When selecting option filter, the selection is reflected inside the URL

### CODE REVIEW

File: COMETwebapp/Components/SystemRepresentation/SystemRepresentationBody.razor.cs [hunk 1]
```diff
@@ -30,39 +36,43 @@ namespace COMETwebapp.Components.SystemRepresentation
...
-        public void OnDomainFilterChange(string domain)
+        private void UpdateUrl()
         {
-            this.ViewModel.OnDomainFilterChange(domain);
+            var additionalParameters = new Dictionary<string, string>();
+
+
+            if (this.ViewModel.OptionSelector.SelectedOption != null)
+            {
+                additionalParameters["option"] = this.ViewModel.OptionSelector.SelectedOption.Iid.ToString();
```
Comment (antoineatrhea) [inline COMETwebapp/Components/SystemRepresentation/SystemRepresentationBody.razor.cs:72]: use the QueryKeys.OptionKey value

File: COMETwebapp/Components/SystemRepresentation/SystemRepresentationBody.razor.cs [hunk 2]
```diff
@@ -30,39 +36,43 @@ namespace COMETwebapp.Components.SystemRepresentation
...
-        ///     Updates Elements list when a filter for domain is selected
+        /// Sets the url of the <see cref="NavigationManager" /> based on the current values
         /// </summary>
-        /// <param name="domain">Name of the selected domain</param>
-        public void OnDomainFilterChange(string domain)
+        private void UpdateUrl()
         {
-            this.ViewModel.OnDomainFilterChange(domain);
+            var additionalParameters = new Dictionary<string, string>();
+
```
Comment (antoineatrhea) [inline COMETwebapp/Components/SystemRepresentation/SystemRepresentationBody.razor.cs:68]: remove additional break line

File: COMETwebapp/ViewModels/Components/SystemRepresentation/SystemRepresentationBodyViewModel.cs [hunk 3]
```diff
@@ -219,10 +188,11 @@ private void CreateElementUsages(IEnumerable<ElementBase> elements)
         /// <param name="parent"></param>
         private void CreateTreeRecursively(ElementBase elementBase, SystemNode current, SystemNode parent)
         {
+            var currentDomain = this.SessionService.GetDomainOfExpertise(this.CurrentIteration);
```
Comment (antoineatrhea) [inline COMETwebapp/ViewModels/Components/SystemRepresentation/SystemRepresentationBodyViewModel.cs:191]: The property CurrentDomain exists, use it instead

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Small changes requested

**Review by antoineatrhea — APPROVED** [review]

**Review by jaimeatrhea — APPROVED** [review]

**Pull request merged**